bftengine/src/bcstatetransfer/BCStateTran.cpp

```diff
@@ -1874,10 +1883,14 @@ bool BCStateTran::onMessage(const ItemDataMsg *m, uint32_t msgLen, uint16_t repl
       return false;
     }
   } else {
-    ConcordAssertEQ(firstRequiredBlock, 0);
-    ConcordAssertEQ(lastRequiredBlock, 0);
+    ConcordAssertEQ(psd_->getFirstRequiredBlock(), 0);
+    ConcordAssertEQ(psd_->getLastRequiredBlock(), 0);
 
-    // if msg is not relevant
+    if (!std::holds_alternative<FetchResPagesMsg>(lastSentMsg_)) {
```

Comment:
I see such code block at some other place.
Should such relevant blocks be called as function?

Reply:
I don't see an adventage since we need a condition anyway to return, we need a return even with a common code, and we need a different error message per case. So in practice we will not save any code. It apears only twice..